packages/sanity/src/_internal/manifest/extractWorkspaceManifest.ts

```diff
@@ -500,3 +515,48 @@ function resolveTitleValueArray(possibleArray: unknown): ManifestTitledValue[] |
 
   return titledValues
 }
+
+const extractManifestTools = (tools: Workspace['tools']): ManifestTool[] =>
+  tools.map((tool) => {
+    const {
+      title,
+      name,
+      icon,
+      __internalApplicationType: type,
+    } = tool as Workspace['tools'][number] & {__internalApplicationType: string}
+    return {
+      title,
+      name,
+      type: type || null,
+      icon: resolveIcon({
+        icon,
+        title,
+      }),
+    } satisfies ManifestTool
+  })
+
+const resolveIcon = (props: SchemaIconProps): string | null => {
+  const sheet = new ServerStyleSheet()
+
+  try {
+    /**
+     * You must render the element first so
+     * the style-sheet above can be populated
+     */
+    const element = renderToString(createElement(SchemaIcon, {...props, sheet}))
```

Comment:
Should the createElement return value here not be wrapped in `sheet.collectStyles()`?

Reply:
it doesn't have to because we're using <StyleSheetManager> but if i'm honest, it would be nicer to just do that.

Reply:
another minor thing, probably doesn't matter much, but could we technically have used `renderToStaticMarkup`?